routes/users.js

```diff
@@ -17,6 +17,7 @@ router.patch("/self", authenticate, userValidator.updateUser, users.updateSelf);
 router.get("/", userValidator.getUsers, users.getUsers);
 router.get("/self", authenticate, users.getSelfDetails);
 router.get("/isUsernameAvailable/:username", authenticate, users.getUsernameAvailabilty);
+router.get("/generateusername", authenticate, users.getUsername);
```

Comment:
I think it would be better if this is a POST route as here we are doing some action. GET is generally used for querying the data and giving it in a specific format

Route can be `/users/username`

Reply:
I want to discuss with you regarding this comment.

Hi @rohan09-raj first of all thank you for reviewing my pr, If you check this RFC After change API contract (<a href=" https://github.com/Real-Dev-Squad/website-api-contracts/issues/148">RFC</a>) I am not modifying the database or creating any new resources when generating a unique username. Instead, I am performing a read-only operation to calculate and provide a username to frontend based on existing data. In such a case, using a GET request in route is consistent with the principles of RESTful API design, where GET requests are used for retrieving resources without making any modifications.

Thank you for route endpoint name